src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs

```diff
@@ -14180,5 +14180,189 @@ .maxstack  1
             verifier.VerifyIL("NativeInts.Checked4", expectedCheckedIL);
             verifier.VerifyIL("NativeInts.Checked5", expectedCheckedIL);
         }
+                             
+        [WorkItem(47887, "https://github.com/dotnet/roslyn/issues/47887")]
+        [Fact]
+        public void ArrayAccessNInt()
+        {
+            string source =
+@"class C
+{
+    static void Main()
+    {
+        int r = F(new int[] { 1 }, 0);
+        System.Console.WriteLine(r);
+    }
+    static int F(int[] a, nint i)
+    {
+        return a[i];
```

Comment:
>i [](start = 17, length = 1)

Are we testing const scenario?

Reply:
Updated to test both const and non-const scenarios for all 6 tests.